onenote/tests/test.bal

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/log;
+import ballerina/os;
+import ballerina/test;
+import ballerina/lang.runtime;
+
+configurable string & readonly refreshUrl = os:getEnv("REFRESH_URL");
+configurable string & readonly refreshToken = os:getEnv("REFRESH_TOKEN");
+configurable string & readonly clientId = os:getEnv("CLIENT_ID");
+configurable string & readonly clientSecret = os:getEnv("CLIENT_SECRET");
+
+Configuration configuration = {
+    authConfig: {
+        clientId: clientId,
+        clientSecret: clientSecret,
+        refreshToken: refreshToken,
+        refreshUrl: refreshUrl
+    }
+};
+
+Client oneNoteClient = check new(configuration);
+string notebookId = EMPTY_STRING;
+string sectionId = EMPTY_STRING;
+string sectionGroupId = EMPTY_STRING;
+string pageId = EMPTY_STRING;
+string pageId2 = EMPTY_STRING;
+
+@test:Config {
+    enable: true
+}
+function testListNotebooks() returns error? {
+    log:printInfo("oneNoteClient->listNotebooks()");
+    Notebook[] notebooks = check oneNoteClient->listNotebooks();
+    log:printInfo("Number of notebooks available: " + notebooks.length().toString());
+    notebookId = notebooks[0].id;
```

Comment:
Don't we need to use assert?

Reply:
No need to assert.
In line no 50, i am accessing the response and getting the first element's notebookId.

if anything went wrong, in line no 48 itself we'll get an error.

And if the array is empty, in line no 50, we'll get an error.

Moreover, using the notebookId taken from this testcase is directly or indirectly used in all other methods.

Reply:
Added assertion to all the test cases